olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp

```diff
@@ -162,10 +162,25 @@ BlobApi::DataResponse DataRepository::GetBlobData(
     }
   }
 
+  // Check if other threads have faced an error.
+  const auto optional_error = mutex.get_error();
```

Comment:
In this case we would need to get rid of std::unique_lock<NamedMutex> and use NamedMutex::unlock() before every return statement. We have 7 return statements in DataRepository::GetBlobData, so it would complicate the code from NamedMutex user perspective.

Reply:
Ok. Let's leave it as you proposed for now, considering that it is a "temporary" solution.